master/internal/logretention/logretention.go

```diff
@@ -0,0 +1,115 @@
+package logretention
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/go-co-op/gocron/v2"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+
+	"github.com/determined-ai/determined/master/internal/db"
+	"github.com/determined-ai/determined/master/pkg/model"
+)
+
+const retainForever = -1
+
+var (
+	log = logrus.WithField("component", "log-retention")
+
+	schedulerDefaultOpts = []gocron.SchedulerOption{
+		gocron.WithLimitConcurrentJobs(1, gocron.LimitModeReschedule),
+	}
+	scheduler gocron.Scheduler
+
+	// WaitGroup is used for testing purposes to wait for the log retention scheduler to finish.
+	WaitGroup *sync.WaitGroup
+)
+
+func init() {
+	SetupScheduler()
```

Comment:
minor style thing, can we rename SetupScheduler - > NewScheduler, have it return the scheduler, then just replace the init block with setting `var scheduler = NewScheduler()` in the block above. and maybe dont export NewScheduler unless we need to.

Reply:
`SetupScheduler` is being called by a test to set the internal `scheduler` variable with a fake clock. Changing it to `NewScheduler` would complicate things and require exposing the internal `scheduler` so tests could create a fake one. This particular flow was chosen to minimize the difference in testing with `master/internal/api_logretention_intg_test.go` vs actual use but there is probably a better way to accomplish this.